internal/controllers/cluster/cluster_controller_status.go

```diff
@@ -622,6 +625,12 @@ func setMachinesReadyCondition(ctx context.Context, cluster *clusterv1.Cluster,
 }
 
 func setMachinesUpToDateCondition(ctx context.Context, cluster *clusterv1.Cluster, machines collections.Machines, getDescendantsSucceeded bool) {
+	// TODO: This should be removed once the UpToDate condition has been implemented for MachinePool Machines
```

Comment:
What about dropping MP machines in updateStatus from  allMachines, workerMachines, machinesToBeRemediated, unhealthyMachines so:
a - we are sure MP & MP machines doesn't show up in any condition / counter
a - we have a single function to change when MP will be ready with v1beta2 status

Reply:
But doesn't it already work in all other cases? MP Machines are also reconciled by the Machine controller

Reply:
What confuses me is that we are dropping MP from everywhere
- replica counters
- Available condition
- ScalingUp condition
- ScalingDown condition

But we keep machine pool machine around, so, in theory those machines can pop up e.g. in machinesReady (or less probably, also in remediating e.g. as unhealthyMachines) machines.

If this happens, this could be confusing, because data at this point are not consistent anymore (e.g. why machinesReady is false with a MPM machine, and e.g replica counters are not showing a machine not ready?)

This is why I think that we should disable MP entirely
But happy to chat about it if you prefer

Reply:
All good, that makes sense